studio/src/ui/popoverMenus/PopoverManager.js

```diff
@@ -119,17 +117,23 @@ export class PopoverManager {
 	}
 
 	#hasBodyClickListener = false;
+
 	/**
 	 * @param {MouseEvent} e
 	 */
 	#onBodyClick = e => {
-		if (this.current && !this.current.el.contains(/** @type {Node} */ (e.target))) {
-			this.closeCurrent();
+		if (this.#activePopovers.length === 0) {
+			throw new Error("Error handling body click: No popovers exist");
 		}
+
+		this.#activePopovers.forEach(p => {
+			p.close();
+		});
 	};
 
 	#updateCurtainActive = () => {
-		if (this.current && this.current.needsCurtain) {
+		const needsCurtains = this.#activePopovers.some(p => p.needsCurtain);
```

Comment:
ok final nit I promise :P
I think this might be a typo and should say `needsCurtain`